packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -568,19 +480,14 @@ function createElementInternalsProxy(
         const vm = getAssociatedVM(this);
         const {
             elm,
-            def: { formAssociated },
             renderer: { attachInternals },
         } = vm;
 
         if (vm.shadowMode === ShadowMode.Synthetic) {
-            throw new Error(
-                'attachInternals API is not supported in light DOM or synthetic shadow.'
-            );
+            throw new Error('attachInternals API is not supported in synthetic shadow.');
```

Comment:
@nolanlawson I gated `attachInternals` behind [API version 61](https://github.com/salesforce/lwc/pull/4035/commits/7b713ae5ff879641786a2cfafaf5d76967e073f5).

I think you're right, the form-associated lifecycle methods are tied to native lifecycle but `ElementInternals` is not. This could mean that a component author could call `attachInternals` on a synthetic lifecycle component, receive the `ElementInternals` object which points to the form.

It should technically not be able to associate the form with synthetic lifecycle enabled, but I don't want to chance running into any bugs in that scenario.

This shouldn't be an issue for `CustomElementConstructor` as it directly inherits from the `HtmlElement` and calls the native `attachInternals` API.

It also only uses native lifecycle so I don't think it should be an issue.

Reply:
This sounds fine to me. Probably no one is relying on older API versions in OSS, so let's just gate all of ElementInternals/FACE behind API version 61.